### PR TITLE
Create Course with CRUD Operations

### PR DESCRIPTION
**Pull request by ySteinhart1**
A few of these commits add a few miscellaneous helper functions. Only the last one directly deals with course CRUD.

### CODE REVIEW

**Review by reznikovl — APPROVED** [review]
LGTM. These changes will be very useful for allowing us to create courses

File: Gemfile.lock [hunk 1]
```diff
@@ -105,6 +105,8 @@ GEM
     net-smtp (0.3.2)
       net-protocol
     nio4r (2.5.8)
+    nokogiri (1.13.9-aarch64-linux)
+      racc (~> 1.4)
```
Comment (ySteinhart1) [inline Gemfile.lock:109]: Not sure why this is added again...

File: app/controllers/authenticated_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class AuthenticatedController < ApplicationController
+    before_action :check_user, :set_user
+
+    def check_user
+        redirect_to root_path if (session[:current_user].nil? || User.find_by(id: session[:current_user]).nil?)
```
Comment (reznikovl) [inline app/controllers/authenticated_controller.rb:5]: nit: is the first part of the if statement necessary? Would find_by just return nil if id is nil?

**Review by reznikovl — APPROVED** [review]
Again looks good, appreciate the authenticated controller

**Pull request merged**